src/vcpkg/vcpkgpaths.cpp

```diff
@@ -845,8 +845,11 @@ If you wish to silence this error and use classic mode, you can:
         std::error_code ec;
         Files::ExclusiveFileLock guard(Files::ExclusiveFileLock::Wait::Yes, fs, lock_file, ec);
 
-        System::Command fetch_git_ref =
-            git_cmd_builder(*this, dot_git_dir, work_tree).string_arg("fetch").string_arg("--").string_arg(repo);
+        System::Command fetch_git_ref = git_cmd_builder(*this, dot_git_dir, work_tree)
+                                            .string_arg("fetch")
+                                            .string_arg("--update-shallow")
```

Comment:
```suggestion
                                            .string_arg("--unshallow")
```
Since git repos reference tree objects in the history, we need the full history available.

Reply:
The point is: For a shallow clone such as `VCPK_INSTALLATION_ROOT` in Azure Pipelines, one would need to unshallow the clone at `VCPKG_INSTALLATION_ROOT`. You cannot apply `--unshallow` when fetching from the shallow clone.

> Since git repos reference tree objects in the history, we need the full history available.

I'm not a git tree object expert. Anyway, do you really always need the full history, or just enough depth to resolve the references for the desired ports and versions?